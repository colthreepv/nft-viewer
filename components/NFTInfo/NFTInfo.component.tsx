import { TokenInfo } from '@/utils/avascan.utils'
import { shortenAddress } from '@/utils/eth.utils'
import { FC } from 'react'

export interface NFTInfoProps {
  data: TokenInfo
}

const NFTInfo: FC<NFTInfoProps> = ({ data }) => {
  // if tokenStandard is not defined, it's ERC721. Otherwise it's either ERC1155 or ERC721
  const tokenStandard =
    data.collection.detail.type == null ? 'ERC721' : data.collection.detail.type === 'erc721' ? 'ERC721' : 'ERC1155'

  return (
    <div>
      <p className="text-lg font-bold text-white">
        {data.collection.name} #{data.tokenId}
      </p>
      <a href="about:blank" target={'_blank'} rel="noreferrer">
        <p className="my-2 text-slate-500 font-bold">{data.collection.name} &gt;</p>
      </a>
      <div className="my-2 py-3 px-2 bg-slate-500 rounded-md text-white">
        <p className="font-extrabold">Details</p>
        <div className="grid grid-cols-[minmax(min-content,_1fr)_max-content] w-full">
          <div className="flex flex-col">
            <p>Owner:</p>
            <p>Contract Address:</p>
            <p>Creator:</p>
            <p>Token ID:</p>
            <p>Token Standard:</p>
          </div>
          <div className="ml-auto gap-y-2 block md:hidden">
            <p>{shortenAddress(data.currentOwner.id)}</p>
            <p>{shortenAddress(data.tokenAddress)}</p>
            <p>{shortenAddress(data.firstOwner.id)}</p>
            <p>{data.tokenId}</p>
            <p>{tokenStandard}</p>
          </div>
          {/* desktop variant */}
          <div className="ml-auto gap-y-2 hidden md:block">
            <p>{data.currentOwner.id}</p>
            <p>{data.tokenAddress}</p>
            <p>{data.firstOwner.id}</p>
            <p>{data.tokenId}</p>
            <p>{tokenStandard}</p>
          </div>
        </div>
      </div>
    </div>
  )
}

export default NFTInfo
