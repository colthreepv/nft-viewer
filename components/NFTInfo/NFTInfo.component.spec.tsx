import { render } from '@testing-library/react'
import NFTInfo, { NFTInfoProps } from './NFTInfo.component'

describe('NFTInfo', () => {
  it('Matches snapshot', async () => {
    const props: NFTInfoProps = {
      data: {
        chainId: '43114',
        tokenAddress: '0xD6F7b2E3D2BE5c12038B24AfCd779e2f179B2400',
        tokenId: '1',
        uriExternal: 'https://infura-ipfs.io/ipfs/bafybeifabq76kkyzozd7j7qncrqwhfham7r5acbinfxfvou7ob6aixtvfm/1.png',
        uri256:
          'https://imgproxy-mainnet.avascan.com/nt2yOaQH3gfeT1HPbP5bXCOjxifqrVRK2FmLDZzTCXA/thumb_256/czM6Ly9hdmFzY2FuLW1haW5uZXQtbmZ0L2VyYzcyMS80MzExNC9kNi9mNy9iMi9lMy9kMmJlNWMxMjAzOGIyNGFmY2Q3NzllMmYxNzliMjQwMC8xLnBuZw',
        uri1024:
          'https://imgproxy-mainnet.avascan.com/mnxeMftCs4-mwNwc9jyj65h50U47fsxVu5qa9ri4jyY/thumb_1024/czM6Ly9hdmFzY2FuLW1haW5uZXQtbmZ0L2VyYzcyMS80MzExNC9kNi9mNy9iMi9lMy9kMmJlNWMxMjAzOGIyNGFmY2Q3NzllMmYxNzliMjQwMC8xLnBuZw',
        imageMd5: 'e972cd46fc9e4bd95cb1019b239484cf',
        createdAt: '2023-01-21T15:01:24.000Z',
        blockNumber: 25230695,
        txHash: '0x95da0205c798ea02473e294d767e55f4420b1dd21dc071d1ec9f58593bf106de',
        firstOwner: { id: '0x2a549e5f473212829d4000305f27529C1Ccbab9A' },
        currentOwner: {
          id: '0x2a549e5f473212829d4000305f27529C1Ccbab9A',
        },
        body: '{}',
        bodyType: 'json',
        collection: {
          address: '0xD6F7b2E3D2BE5c12038B24AfCd779e2f179B2400',
          name: 'SayCheese By KimiNaki',
          symbol: 'CHESEE',
          detail: {},
        },
      },
    }

    const rendered = render(<NFTInfo {...props} />)

    expect(rendered).toMatchSnapshot()

    const { findByText } = rendered
    const contractAddress = await findByText(props.data.tokenAddress)

    expect(contractAddress).toBeInTheDocument()
  })
})
