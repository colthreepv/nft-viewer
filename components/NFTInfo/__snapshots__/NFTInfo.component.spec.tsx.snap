// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NFTInfo Matches snapshot 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div>
        <p
          class="text-lg font-bold text-white"
        >
          SayCheese By KimiNaki
           #
          1
        </p>
        <a
          href="about:blank"
          rel="noreferrer"
          target="_blank"
        >
          <p
            class="my-2 text-slate-500 font-bold"
          >
            SayCheese By KimiNaki
             &gt;
          </p>
        </a>
        <div
          class="my-2 py-3 px-2 bg-slate-500 rounded-md text-white"
        >
          <p
            class="font-extrabold"
          >
            Details
          </p>
          <div
            class="grid grid-cols-[minmax(min-content,_1fr)_max-content] w-full"
          >
            <div
              class="flex flex-col"
            >
              <p>
                Owner:
              </p>
              <p>
                Contract Address:
              </p>
              <p>
                Creator:
              </p>
              <p>
                Token ID:
              </p>
              <p>
                Token Standard:
              </p>
            </div>
            <div
              class="ml-auto gap-y-2 block md:hidden"
            >
              <p>
                0x2a54...ab9A
              </p>
              <p>
                0xD6F7...2400
              </p>
              <p>
                0x2a54...ab9A
              </p>
              <p>
                1
              </p>
              <p>
                ERC721
              </p>
            </div>
            <div
              class="ml-auto gap-y-2 hidden md:block"
            >
              <p>
                0x2a549e5f473212829d4000305f27529C1Ccbab9A
              </p>
              <p>
                0xD6F7b2E3D2BE5c12038B24AfCd779e2f179B2400
              </p>
              <p>
                0x2a549e5f473212829d4000305f27529C1Ccbab9A
              </p>
              <p>
                1
              </p>
              <p>
                ERC721
              </p>
            </div>
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div>
      <p
        class="text-lg font-bold text-white"
      >
        SayCheese By KimiNaki
         #
        1
      </p>
      <a
        href="about:blank"
        rel="noreferrer"
        target="_blank"
      >
        <p
          class="my-2 text-slate-500 font-bold"
        >
          SayCheese By KimiNaki
           &gt;
        </p>
      </a>
      <div
        class="my-2 py-3 px-2 bg-slate-500 rounded-md text-white"
      >
        <p
          class="font-extrabold"
        >
          Details
        </p>
        <div
          class="grid grid-cols-[minmax(min-content,_1fr)_max-content] w-full"
        >
          <div
            class="flex flex-col"
          >
            <p>
              Owner:
            </p>
            <p>
              Contract Address:
            </p>
            <p>
              Creator:
            </p>
            <p>
              Token ID:
            </p>
            <p>
              Token Standard:
            </p>
          </div>
          <div
            class="ml-auto gap-y-2 block md:hidden"
          >
            <p>
              0x2a54...ab9A
            </p>
            <p>
              0xD6F7...2400
            </p>
            <p>
              0x2a54...ab9A
            </p>
            <p>
              1
            </p>
            <p>
              ERC721
            </p>
          </div>
          <div
            class="ml-auto gap-y-2 hidden md:block"
          >
            <p>
              0x2a549e5f473212829d4000305f27529C1Ccbab9A
            </p>
            <p>
              0xD6F7b2E3D2BE5c12038B24AfCd779e2f179B2400
            </p>
            <p>
              0x2a549e5f473212829d4000305f27529C1Ccbab9A
            </p>
            <p>
              1
            </p>
            <p>
              ERC721
            </p>
          </div>
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
